### PR TITLE
fix: Explore misleading save action

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
When saving a chart in Explore, we have different states:
- If a user is overriding a chart or creating a new one
- If a user is selecting a dashboard or creating a new one
- If a user can override a chart or not

Currently, the Save modal buttons change to accommodate different combination of states:
- Save
- Save as new chart
- Save to new dashboard
- Save & go to dashboard
- Save & go to new dashboard

The problem is that are missing some combinations which leads to confusion when saving charts:
- Save as new chart to new dashboard
- Save as new chart & go to dashboard
- Save as new chart & go to new dashboard

One example of such confusion is when a user is not an owner of a chart but can save that chart as a new one. The button displays Save & go to dashboard which leads the user to think that a chart will be overridden in that dashboard but what actually happens is that a new chart is created and added to the dashboard. One can argue that the Save as radio selection would be sufficient to indicate the difference but that's not the case. Specially because we're changing the buttons text for some states and not for others. 

To fix this, my first attempt was to make sure the button labels accounted for all possible states but that resulted in a poor UI as you can see bellow:

Screenshot 2023-08-01 at 13 11 06

<br>A better solution was to change the modal to preserve the button labels and add an information message when creating a new chart, a dashboard or both:

Screenshot 2023-08-01 at 15 48 24

<br>This message only appears when it's not clear that a new asset will be created. If a user is creating a chart for the first time and adding it to an existing dashboard, no message is displayed.

This PR also moves the alert message to the bottom, similar to the native filters modal, closer to the dashboard select and save button which are the sources for the alert.

Screenshot 2023-08-01 at 15 54 18

### TESTING INSTRUCTIONS
Make sure the correct information message is displayed or not depending on the chart state.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 1]
```diff
@@ -324,89 +320,115 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
...
+              allowClear
+              allowNewOptions
+              ariaLabel={t('Select a dashboard')}
+              options={this.loadDashboards}
+              onChange={this.onDashboardChange}
+              value={this.state.dashboard}
+              placeholder={
+                <div>
+                  <b>{t('Select')}</b>
+                  {t(' a dashboard OR ')}
```
Comment (justinpark) [inline superset-frontend/src/explore/components/SaveModal.tsx:391]: These labels should be a good example of adding unit tests.

**Review by kasiazjc — APPROVED** [review]
I think this change makes the process more intuitive, so lgtm design wise. Thanks!

File: superset-frontend/src/explore/components/SaveModal.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/explore/components/SaveModal.tsx:391]: Yes but I'll focus on the PR changes. This part was only indented.

**Review by justinpark — APPROVED** [review]
LGTM

**Pull request merged**